frontend/src/bundles/common/components/button/button.tsx

```diff
@@ -1,15 +1,51 @@
-type Properties = {
-    label: string;
-    type?: 'button' | 'submit';
+import { getValidClassNames } from '~/bundles/common/helpers/helpers.js';
+import { type ButtonProperties } from '~/bundles/common/types/button.type.js';
+
+const classes = {
```

Comment:
let's split this into maps, like buttonVariantToClasses: Record<ButtonVariant, string> and inline base classes.
Also common components should have w-full